src/java.desktop/share/classes/javax/swing/border/EtchedBorder.java

```diff
@@ -131,22 +151,47 @@ public EtchedBorder(int etchType, Color highlight, Color shadow)    {
      * @param height the height of the painted border
      */
     public void paintBorder(Component c, Graphics g, int x, int y, int width, int height) {
-        int w = width;
-        int h = height;
+        // We remove any initial transforms to prevent rounding errors
+        // when drawing in non-integer scales
+        AffineTransform at = new AffineTransform();
+        Stroke oldStk = new BasicStroke();
+        int stkWidth = 1;
+        if (g instanceof Graphics2D) {
+            Graphics2D g2d = (Graphics2D) g;
+            at = g2d.getTransform();
+            oldStk = g2d.getStroke();
+            if((at.getType() & AffineTransform.TYPE_MASK_SCALE) != 0) {
```

Comment:
if((at.getType() & AffineTransform.TYPE_MASK_SCALE) != 0)

getType() returns a mask of set bits. 

You don't want to check that SCALE is set, you want to check that ROTATION and TRANSLATE are NOT set.

Something like 

if (at.getType() & ~(TYPE_MASK_TRANSLATION | TYPE_MASK_ROTATION) == 0)) {